pkg/controller/util/channel.go

```diff
@@ -52,10 +52,11 @@ func NewChannelOwnerReference(channel *kafkav1beta1.KafkaChannel) metav1.OwnerRe
 func ChannelDnsSafeName(kafkaSecretName string) string {
 
 	// In order for the resulting name to be a valid DNS component it's length must be no more than 63 characters.
-	// We are consuming 9 chars for the component separators, and the Channel suffix, which reduces the available
-	// length to 54. We will allocate 50 characters to the kafka secret name leaving an extra buffer.
-	safeSecretName := GenerateValidDnsName(kafkaSecretName, 50, true, false)
-	return fmt.Sprintf("%s-channel", safeSecretName)
+	// We are consuming 10 chars for the component separators, 8 for the hash, and the Channel suffix, which reduces the
+	// available length to 45. We will allocate 41 characters to the kafka secret name leaving an extra buffer.
+	safeSecretName := GenerateValidDnsName(kafkaSecretName, 41, true, false)
```

Comment:
This comment is not right - the wording was "10 for the separator and channel suffix" and you squeezed the hash in there so not it sounds like the separators by themselves take 10 and the channel suffix is ?  Please reword it.

Reply:
You're right, that was clumsy (and I did the math wrongly to boot).  Fixed the numbers and sentence structure.